src/main/cpp/subsystems/LimeLightUtil.cpp

```diff
@@ -31,12 +33,33 @@ void t34::LimelightUtil::Periodic()
     m_table = LimelightHelpers::getLimelightNTTable(LIMELIGHT_TABLE_NAME);
     m_tx = LimelightHelpers::getTX(LIMELIGHT_TABLE_NAME);
     m_ty = LimelightHelpers::getTY(LIMELIGHT_TABLE_NAME);
+    m_current_id = LimelightHelpers::getFiducialID(LIMELIGHT_TABLE_NAME);
+
+    switch (m_target_mode)
+    {
+        case (TargetMode::kAmp):
+            m_math_handler.SetTargetHeightMeters(0.864);
+            
+            m_target_id = (m_current_id == 5 || m_current_id == 6) ? m_current_id : -1.0;
+
+            break;
+        
+        case (TargetMode::kSpeaker):
+            m_math_handler.SetTargetHeightMeters(1.9815);
+
+            m_target_id = ( m_current_id == 1 ||
+                            m_current_id == 2 ||
+                            m_current_id == 3 ||
+                            m_current_id == 4) ? m_current_id : -1.0;
```

Comment:
It looks as if AprilTags 3 & 4 label the Blue Alliance's **source**. The Blue Alliance speaker has AprilTags 7 & 8:

```suggestion
            m_target_id = ( m_current_id == 1 ||
                            m_current_id == 2 ||
                            m_current_id == 7 ||
                            m_current_id == 8) ? m_current_id : -1.0;
```